engine/flink/process/src/main/scala/pl/touk/nussknacker/engine/process/FlinkStreamingProcessRegistrar.scala

```diff
@@ -240,18 +242,22 @@ class FlinkStreamingProcessRegistrar(compileProcess: (EspProcess, ProcessVersion
   }
 
   private def configureCheckpoints(env: StreamExecutionEnvironment, streamMetaData: StreamMetaData): Unit = {
-    val checkpointIntervalToSetInMillis = streamMetaData.checkpointIntervalDuration.getOrElse(checkpointInterval).toMillis
-    env.enableCheckpointing(checkpointIntervalToSetInMillis)
+    val processSpecificCheckpointIntervalDuration = streamMetaData.checkpointIntervalDuration
+    val checkpointIntervalToSetInMillis = processSpecificCheckpointIntervalDuration.getOrElse(checkpointConfig.checkpointInterval).toMillis
 
-    //TODO: should this be configurable?
-    env.getCheckpointConfig.setMinPauseBetweenCheckpoints(checkpointIntervalToSetInMillis / 2)
-    env.getCheckpointConfig.setMaxConcurrentCheckpoints(1)
+    env.enableCheckpointing(checkpointIntervalToSetInMillis)
+    env.getCheckpointConfig.setMinPauseBetweenCheckpoints(checkpointConfig.minPauseBetweenCheckpoints.map(_.toMillis).getOrElse(checkpointIntervalToSetInMillis / 2))
```

Comment:
Can you replace this logic with:
```
checkpointConfig.minPauseBetweenCheckpoints.map(_.toMillis).foreach(env.getCheckpointConfig.setMinPauseBetweenCheckpoints)
```
?
Thanks to that won't be need this `getOrElse(checkpointIntervalToSetInMillis / 2)` with magic numbers - let's use flink's defaults

Reply:
I will keep `checkpointIntervalToSetInMillis / 2` to make NK compatible with previous versions